glean-core/ios/Glean/Glean.swift

```diff
@@ -111,6 +111,11 @@ public class Glean {
         if isInitialized() {
             glean_set_upload_enabled(handle, enabled.toByte())
         }
+
+        // Cancel background metric recording and upload tasks
+        if !uploadEnabled {
```

Comment:
In #288 this function is changed. We probably should land that first, then change it here to cancel background tasks before we disable stuff.

Reply:
I can leave this out for now